src/executor.ts

```diff
@@ -383,6 +477,12 @@ class ScriptExecution {
     }
 
     return this.#workerPool.run(async (): Promise<Result<void>> => {
+      // Significant time could have elapsed since we last checked because of
+      // parallelism limits.
+      if (this.#shouldCancel) {
+        return {ok: false, error: this.#cancelledEvent};
```

Comment:
Some of the sites are places where we return `ExecutionResult`s, which are `Result<ScriptState, Failure[]>`. That's the top-level thing we return from execution, and it's got an array of `Failure`s because when there are multiple dependencies, you might have multiple errors.

Other sites are more drilled-in functions, which just return a specific single error if they fail. Actually the `#acquireLock` one should probably be like that, done.